### DISCUSSION TITLE
How to set a particular cuda device for auto3dseg (algo.train())?

### DISCUSSION THREAD

**Discussion by rajanish4** [Project-MONAI/tutorials#1538] [Q&A]
Hi, I am currently running auto3dseg by first using BundleGen.generate and then i use the algo.train() method to start all networks sequentially. My problem is that the program always runs on device 0 even if i set os.environment['CUDA_VISIBLE_DEVICES'] = '3' or any other number (I have 4 gpus). Can you please help me with setting the gpu for autorunner?

**Comment by KumoLiu** [comment]
Hi @rajanish4, could you please try `set_device_info` in `AutoRunner`?
https://github.com/Project-MONAI/MONAI/blob/14fcf72a6b733d7b65888d67181fd66d8ebcf755/monai/apps/auto3dseg/auto_runner.py#L477

**Comment by rajanish4** [comment]
Hi @KumoLiu Thanks for the information, I am new to autorunner and not really sure where to change this. I am using this tutorial from Monai: https://github.com/Project-MONAI/tutorials/blob/main/auto3dseg/notebooks/auto3dseg_autorunner_ref_api.ipynb 
Should i train something with the algo.train() function in this notebook ? Can you tell me in regards to this notebook please. Thanks in advance

**Reply by KumoLiu** [reply]
Hi @rajanish4, you can set `set_device_info` after initialize AutoRunner `runner = AutoRunner(input=input)`. Thanks!
